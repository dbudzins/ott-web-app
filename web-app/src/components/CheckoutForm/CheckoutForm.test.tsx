import React from 'react';
import { render } from '@testing-library/react';

import offer from '../../fixtures/monthlyOffer.json';
import order from '../../fixtures/order.json';
import type { Offer, Order } from '../../../types/checkout';

import CheckoutForm from './CheckoutForm';

describe('<CheckoutForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <CheckoutForm
        onCloseCouponFormClick={vi.fn()}
        onCouponFormSubmit={vi.fn()}
        onRedeemCouponButtonClick={vi.fn()}
        onCouponInputChange={vi.fn()}
        onBackButtonClick={vi.fn()}
        onPaymentMethodChange={vi.fn()}
        couponFormOpen={false}
        couponFormApplied={false}
        couponInputValue=""
        couponFormError={undefined}
        couponFormSubmitting={false}
        order={order as Order}
        offer={offer as Offer}
        offerType={'svod'}
        submitting={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
