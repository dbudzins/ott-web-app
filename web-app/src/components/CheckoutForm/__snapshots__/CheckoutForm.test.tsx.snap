// Vitest Snapshot v1

exports[`<CheckoutForm> > renders and matches snapshot 1`] = `
<div>
  <div>
    <div
      aria-label="Go back"
      class="iconButton dialogBackButton"
      role="button"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="icon"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          />
        </g>
      </svg>
    </div>
    <h2
      class="title"
    >
      checkout.payment_method
    </h2>
    <div
      class="order"
    >
      <div
        class="orderInfo"
      >
        <p
          class="orderTitle orderTitleMargin"
        >
          checkout.monthly
        </p>
      </div>
      <div
        class="orderPrice"
      >
        <span>
          € 6,99
        </span>
        <small>
          /
          periods.month
        </small>
      </div>
    </div>
    <div
      class="couponForm"
    >
      <button
        class="button default outlined"
      >
        <span>
          checkout.redeem_coupon
        </span>
      </button>
    </div>
    <div>
      <table
        class="orderTotals"
      >
        <tbody>
          <tr>
            <td>
              checkout.payment_method_fee
            </td>
            <td>
              € 0,00
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td>
              checkout.total_price
            </td>
            <td>
              € 6,99
            </td>
          </tr>
          <tr>
            <td>
              checkout.applicable_tax
            </td>
            <td>
              € 1,21
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
    <hr
      class="divider"
    />
    <div
      class="paymentMethods"
    />
    <div />
  </div>
</div>
`;
